sdonboarding.client/src/components/UITemplate/DisplayDataTemplate.jsx

```diff
@@ -0,0 +1,133 @@
+﻿/**
+ * This is a template file for Displaying an entity.
+ * Used by Customer, Product, Store, Sale Pages.
+ * 'type' should be passed to this component for it to fetch the required information.
+ */
+
+import PropTypes from 'prop-types';
+import { useGetDisplayObject, useSetEditObject, useSetDeleteObject, useGetSortListAsc, useGetSortListDesc } from '../../hooks/DisplayDataHooks';
+import { genericDisplayDataMethods } from '../../utils/GenericDisplayMethods';
+import { PencilSquareIcon, TrashIcon } from '@heroicons/react/24/outline';
+import { useCallback, useEffect, useState } from 'react';
+import '../../styles/display.css';
+
+const DisplayTableTemplate = ({ type }) => {
+
+    //Used to sort the displayed Items
+    const [sort, setSort] = useState(['asc', 'id']);
+
+    /* custom hooks for edit delete and fetching objects to be displayed */
+    const setEditDisplay = useSetEditObject(type);
+    const setDeleteDisplay  = useSetDeleteObject(type);
+    const [objects, setObjects] = useState(useGetDisplayObject(type));
+    const getSortListAsc = useGetSortListAsc(type);
+    const getSortListDesc = useGetSortListDesc(type);
+    
+
+    /* get the table column title and the object keys that we want to display on screen*/
+    const theader = genericDisplayDataMethods.getDisplayTableHeader(type);
+
+    console.log("object loaded");
+
+    const handleEdit = (object) => {
+        setEditDisplay(object);
+    };
+
+    const handleDelete = (object) => {
+        setDeleteDisplay(object);
+    };
+
+    const handleSortAsc = (key) => {
+        setSort(["asc", key]);
+    };
+
+    const handleSortDesc = (key) => {
+        setSort(["desc", key]);
+    };
+
+    const sortList = useCallback(() => {
+        if (sort[0] !== 'asc' || sort[1] !== 'id') {
+            if (sort[0] === 'asc') {
+                const sortedData = getSortListAsc(sort[1]);
+                setObjects(sortedData);
+            }
```

Comment:
How sure are you that sort will always have an item? What will happen if **sort** accidentally contains an object instead of array? Again, validate and verify first if **sort** is actually an array and has values before targeting any specific index